components/gh_events/GithubFeed.tsx

```diff
@@ -0,0 +1,102 @@
+"use client";
+import GitHubEvent from "@/components/gh_events/GitHubEvent";
+import { IGitHubEvent } from "@/lib/gh_events";
+import { useState } from "react";
+
+interface Filter {
+  title: string;
+  options: string[];
+}
+interface Props {
+  events: IGitHubEvent[];
+  filterEvetns: Filter[];
+}
+
+const GithubFeed = (props: Props) => {
+  const { events, filterEvetns } = props;
+  const [e, setEvents] = useState(events);
+  const [repo, setRepo] = useState("All");
+  const [eventType, setEventType] = useState("All");
```

Comment:
Use URL search params for storing the filter states instead of using a useState. This way we can also make it a SSR function.

Reply:
> Use URL search params for storing the filter states instead of using a useState. This way we can also make it a SSR function.

Okay got it I will use search params but, We are using onChange and onClick so we can not make SSR right, We have to use "use client"

Reply:
onChange and onClick should update the query params instead.

Reply:
The buttons alone can be a client component. not the entire page